activerecord/test/cases/associations/inverse_associations_test.rb

```diff
@@ -64,6 +65,15 @@ def test_has_many_and_belongs_to_should_find_inverse_automatically
     assert_equal rating_reflection, comment_reflection.inverse_of, "The Comment reflection's inverse should be the Rating reflection"
   end
 
+  def test_has_many_and_belongs_to_should_find_inverse_automatically_for_extension_block
+    comment_reflection = Comment.reflect_on_association(:post)
+    post_reflection = Post.reflect_on_association(:comments)
+
+    assert_predicate post_reflection, :has_inverse?
+    assert_equal comment_reflection, post_reflection.inverse_of
+    assert_equal({}, post_reflection.options)
```

Comment:
I changed this PR to keep the previous behavior (don't mutate `options[:extend]`) and added a test case.